doc/developers-guide/Hooks-and-handlers.md

```diff
@@ -23,50 +23,58 @@ mod_offline:store_packet(From, To, Packet)
 
 Note that in this example `ejabberd_sm` is coupled with `mod_offline`.
 I.e. if `mod_offline` was not available, the code would simply crash; if it was misconfigured or turned off, the behaviour would be undefined.
-To avoid that coupling and also to enable other ([possibly yet to be written](#sidenote-code-yet-to-be-written)) code to carry out some action at this particular moment, `ejabberd_sm` would instead call:
+To avoid that coupling and also to enable other ([possibly yet to be written](#sidenote-code-yet-to-be-written)) code to carry out some action at this particular moment, `ejabberd_sm` calls instead:
 
 ```erlang
-Acc1 = ejabberd_hooks:run_fold(offline_message_hook,
-                               HostType,
-                               Acc,
-                               [From, To, Packet])
+mongoose_hooks:offline_message_hook(Acc, From, To, Packet);
 ```
 
+`mongoose_hooks` is a module which serves as an API for calling all hooks in the server.
+For every hook, there needs to be a function in this module written beforehand which accepts the correct arity of arguments and makes the call to actual low-level hooks mechanism.
+This means that there is some degree of coupling still - but this time between the `ejabberd_sm` module and `mongoose_hooks`, and the latter is always available.
+
 The extra level of indirection introduced by this call gives the flexibility to determine at runtime what code actually gets run at this point.
-`offline_message_hook` is just the name of the hook (in other words of the event that is being signalled);
-`From`, `To` and `Packet` are the arguments passed to the handler just as they would in case of the function being called directly;
-`HostType` is [the host type of the XMPP domain for which this hook is signalled](#sidenote-multiple-domains).
+This depends on which handlers are registered to process the event.
+
+`offline_message_hook` is the name of the hook (in other words of the event that is being signalled);
+`Acc` is the [Accumulator, described later](#using-accumulators);
+`From`, `To` and `Packet` are the arguments passed to the handler, just as they would in case of the function being called directly.
+
+!!! note "Why do we even need the `mongoose_hooks` module?"
```

Comment:
```suggestion
!!! Note "Why do we even need the `mongoose_hooks` module?"
```

Reply:
This should be rendered like this no matter if it is "note" or "Note":
![obraz](https://user-images.githubusercontent.com/34194983/134878403-549510e3-33c0-4cc8-874c-afe414ed0491.png)

If you think it should be changed, we can of course change it. I just followed the examples from here: https://squidfunk.github.io/mkdocs-material/reference/admonitions/#changing-the-title

Reply:
Yes I suppose if we use `!!! Note` in all places we need to do it here to, or change to lowercase in all other places

Reply:
OK, I see we use "Note" instead of "note" in other places, so let's change this to be consistent.